### DISCUSSION TITLE
EPERM: operation not permitted, mkdir

### DISCUSSION THREAD

**Discussion by Tanggue** [aclap-dev/video-downloadhelper#140] [Help and Q&A]
### Your question

When I click on the object I want to download in the tab, the following message appears and an error occurs and the download cannot be performed.
How can I solve this???
(When the download starts normally, a number appears in a blue circle, but an error occurs and a red circle appears.)

```

EPERM: operation not permitted, mkdir 'D:\'

moz-extension://a711294d-3ca0-44f7-9034-4dd69b335836/background/main.js:1
receive@moz-extension://a711294d-3ca0-44f7-9034-4dd69b335836/background/main.js:1:13157
Jp</callCatchAppNotFound/</<@moz-extension://a711294d-3ca0-44f7-9034-4dd69b335836/background/main.js:1:42564

![2](https://github.com/aclap-dev/video-downloadhelper/assets/160303209/ddff2819-cc0e-4933-9314-36dac4e4cfcf)
![3](https://github.com/aclap-dev/video-downloadhelper/assets/160303209/9085c369-d5ba-4e3f-8fc2-7772b78f1405)

```

**Comment by paulrouget** [comment]
I think I fixed that in an upcoming update. Wait a day or two, should be fixed soon.

If you use Firefox, you can try the beta: https://downloadhelper.net/beta

**Reply by Tanggue** [reply]
Thank you for quick response.

**Reply by Fabrice35** [reply]
Hello,
I've had the same error for a few days now.
With chrome or firefox version 8.2.1.3a1 (beta), it's the same thing.
moz-extension://a43a3c38-dcb0-49d7-a5fb-72e52ba9a4c4/background/main.js:1 receive@moz-extension://a43a3c38-dcb0-49d7-a5fb-72e52ba9a4c4/background/main.js:1:13157 Zp</callCatchAppNotFound/</<@moz-extension://a43a3c38-dcb0-49d7-a5fb-72e52ba9a4c4/background/main.js:1:42626

Chrome:
Error: Exit code: 1 at rs.receive (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:13157) at chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:42624

**Reply by paulrouget** [reply]
Ah, damn. I'll work on that asap!

**Reply by paulrouget** [reply]
@Fabrice35 which operating system do you use? And what's the path of the download directory you're setting?

**Reply by paulrouget** [reply]
@Fabrice35 when does it happen exactly?

**Reply by Fabrice35** [reply]
Distributor ID:	Linuxmint
Description:	LMDE 5 (elsie)
Release:	5
Codename:	elsie

Dist based on debian 11.2

Kernel 5.10.0-28-amd64 #1 SMP Debian 5.10.209-2 (2024-01-31) x86_64 GNU/Linux

Tested on chrome Version 121.0.6167.160 (Build officiel) (64 bits)
and firefox 122.0.1 (64 bits)

**Reply by Fabrice35** [reply]
I use Video DownloadHelper Version 8.2.1.1 (stable) and  VdhCoApp 2.0.19
I tried the curl installation method and also with the deb file:
$ /opt/vdhcoapp/vdhcoapp --info
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/xxxxxx",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"

I get the same error when I try to download this video:
https://www.6play.fr/notre-ecole-en-10-questions-p_25421/notre-ecole-en-10-questions-c_13056192

"
Exit code: 1
moz-extension://1d6f97a9-066e-42f8-9995-8393effba76f/background/main.js:1
receive@moz-extension://1d6f97a9-066e-42f8-9995-8393effba76f/background/main.js:1:13157
Zp</callCatchAppNotFound/</<@moz-extension://1d6f97a9-066e-42f8-9995-8393effba76f/background/main.js:1:42626
"

**Reply by paulrouget** [reply]
Ok, I'm getting other reports about EPerm error. Looking at that now.

**Reply by paulrouget** [reply]
If you use Firefox, mind trying the beta version of the addon. I think it might have fixed the issue: https://www.downloadhelper.net/beta (after install, check that it's version 8.2.1.4).

**Reply by Fabrice35** [reply]
It works!
Thanks

Video DownloadHelper 8.2.1.5a1 (beta)

**Reply by Fabrice35** [reply]
Sorry, it worked once, but the resulting video was not watchable.
And when I tried to download it again, I got the same error.

**Reply by GrampaWildWilly** [reply]
Unwatchable . . .  That may mean your content is protected by DRM.  Search this forum for other mentions of DRM.  If you disable DRM in the browser settings, then reload the web page, then try to play the content in the player on the web page, what happens?

**Reply by paulrouget** [reply]
> the resulting video was not watchable.

Link?

**Reply by Fabrice35** [reply]
https://we.tl/t-oG2kyzm2T4

**Comment by Fabrice35** [comment]
> Unwatchable . . . That may mean your content is protected by DRM. Search this forum for other mentions of DRM. If you disable DRM in the browser settings, then reload the web page, then try to play the content in the player on the web page, what happens?

If I deactivate DRM, the video can no longer be played.

**Reply by paulrouget** [reply]
Yeah, so sadly nothing we can do about that.

**Reply by Fabrice35** [reply]
OK, thanks for your reply.
I'm switching to SimpleScreenRecorder for DRM-protected videos.
It's a lot less easy, but the rendering is pretty good!
